core/types/gen_header_json.go

```diff
@@ -30,6 +31,7 @@ func (h Header) MarshalJSON() ([]byte, error) {
 		Extra       hexutil.Bytes  `json:"extraData"        gencodec:"required"`
 		MixDigest   common.Hash    `json:"mixHash"          gencodec:"required"`
 		Nonce       BlockNonce     `json:"nonce"            gencodec:"required"`
+		BaseFee     *hexutil.Big   `json:"baseFeePerGas" rlp:"optional"`
```

Comment:
do we need to mention `optional`? I think default is optional

Reply:
The default value of this tag is `false`.

```go
type field struct {
	optional bool
}

func parseTag(field Field, lastPublic int) (Tags, error) {
		case "optional":
			ts.Optional = true
```